### PR TITLE
kos' day 2,3,4

### PR DESCRIPTION
**Pull request by kand**

### CODE REVIEW

File: 2022/kand/d4.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+print(sum([1 if not (r1 := set(range(int((r1s := (ag := l.split(','))[0].split('-'))[0]), int(r1s[1]) + 1))) - (r2 := set(range(int((r2s := ag[1].split('-'))[0]), int(r2s[1]) + 1))) or not r2 - r1 else 0 for l in open('d4.in')]))
```
Comment (westeezy) [inline 2022/kand/d4.py:1]: this is a file with text. can confirm that much is definitely true

File: 2022/kand/d2-1.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    for line in f:
+        a, b = line.replace('\n', '').split(' ')
+
+        ta = t(a)
+        tb = t(b)
+
+        print(ta, tb)
+        roundScore = 3 if ta == tb else 0
+
+        if tb == 'R':
```
Comment (westeezy) [inline 2022/kand/d2-1.py:20]: I like how you solved it programatically. i was sitting here like theres only N outcomes of this game so I'll just hardcode it all. see #5  LOL.

**Comment by westeezy** [comment]
I didn't do day 3 or 4 yet. Saving those for tomorrow so I stopped peeking after day2

File: 2022/kand/d4.py (hunk 1 above)
Comment (kand) [reply 2022/kand/d4.py:1]: I've been having fun playing some code golf

File: 2022/kand/d2-1.py (hunk 2 above)
Comment (kand) [reply 2022/kand/d2-1.py:20]: Yea my code golf-ed solution for part 2 ended up there pretty much. See `d2-2-2.py` below.

**Pull request merged**